pkg/controller/worker/machines.go

```diff
@@ -286,3 +294,39 @@ func computeEBSDeviceNameForIndex(index int) (string, error) {
 
 	return deviceNamePrefix + deviceNameSuffix[index:index+1], nil
 }
+
+func computeAdditionalHashData(pool extensionsv1alpha1.WorkerPool, workerConfig *awsapi.WorkerConfig) []string {
+	var additionalData []string
+
+	if pool.Volume != nil && pool.Volume.Encrypted != nil {
+		additionalData = append(additionalData, strconv.FormatBool(*pool.Volume.Encrypted))
+	}
+
+	for _, dv := range pool.DataVolumes {
+		additionalData = append(additionalData, dv.Size)
+
+		if dv.Type != nil {
+			additionalData = append(additionalData, *dv.Type)
+		}
+
+		if dv.Encrypted != nil {
+			additionalData = append(additionalData, strconv.FormatBool(*dv.Encrypted))
+		}
+	}
+
+	if workerConfig.Volume != nil && workerConfig.Volume.IOPS != nil {
+		additionalData = append(additionalData, strconv.FormatInt(*workerConfig.Volume.IOPS, 10))
+	}
+
+	for _, dv := range workerConfig.DataVolumes {
+		if dv.IOPS != nil {
+			additionalData = append(additionalData, strconv.FormatInt(*dv.IOPS, 10))
+		}
+
+		if dv.SnapshotID != nil {
+			additionalData = append(additionalData, *dv.SnapshotID)
+		}
+	}
```

Comment:
Hmmm, the worker provider config is already considered in `worker.WorkerPoolHash`. Is it needed to also add it to the additional hash data?

Reply:
Yes, indeed, nice catch. Removed the code, thanks.